lib/sinatra/base.rb

```diff
@@ -348,7 +348,7 @@ def attachment(filename = nil, disposition = :attachment)
       response['Content-Disposition'] = disposition.to_s
       if filename
         params = '; filename="%s"' % File.basename(filename)
-        response['Content-Disposition'] << params
+        response['Content-Disposition'] += params
```

Comment:
Here as well, I'm not sure this change is necessary as we haven't seen any failures in this case